### DISCUSSION TITLE
Is it possible to create a new User-Defined Identity Provider type that allows multiple instances?

### DISCUSSION THREAD

**Discussion by xgp** [keycloak/keycloak#29666] [Q&A] [answered]
I have a new generic identity provider type that I've implemented by extending `AbstractIdentityProvider` and `AbstractIdentityProviderFactory`. The identity provider is **NOT** a "social" identity provider, but an implementation of a protocol (similar to OIDC) that is commonly used by enterprise IAM systems. Thus it is necessary to allow multiple instances of this generic type for each Realm.

For this identity provider, I want:
1. It to appear in the Admin UI in the section with "User-defined" identity provider types. :+1: : This works as expected.
2. To allow the user to specify an `alias`, as in the SAML/OIDC/KeycloakOIDC types, so that users can create multiple instances. :-1: I am unable to determine if this is possible, based on a review of the similar, [user-defined types](https://github.com/keycloak/keycloak/tree/main/services/src/main/java/org/keycloak/broker), and the [Admin UI code](https://github.com/keycloak/keycloak/tree/main/js/apps/admin-ui/src/identity-providers) that renders the IdP section. **Does anyone know if this is possible, and if so, how to achieve it?**

Thank you!

**Comment by pedroigor** [comment, marked as answer]
This one should work after https://github.com/keycloak/keycloak/pull/30208. So you should see the `alias ` and `displayName` in the UI. Therefore, allowed to create multiple instances of the broker.
